src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaOauthSecurityConfig.java

```diff
@@ -45,121 +75,162 @@
 public class IridaOauthSecurityConfig {
 	private static final Logger logger = LoggerFactory.getLogger(IridaOauthSecurityConfig.class);
 
-	@Bean
-	@Primary
-	public ResourceServerTokenServices tokenServices(@Qualifier("clientDetails") ClientDetailsService clientDetails,
-			@Qualifier("iridaTokenStore") TokenStore tokenStore) {
-		DefaultTokenServices services = new DefaultTokenServices();
-		services.setTokenStore(tokenStore);
-		services.setSupportRefreshToken(true);
-		services.setClientDetailsService(clientDetails);
-		return services;
-	}
-
-	@Bean
-	public ClientDetailsUserDetailsService clientDetailsUserDetailsService(
-			@Qualifier("clientDetails") ClientDetailsService clientDetails) {
-		ClientDetailsUserDetailsService clientDetailsUserDetailsService = new ClientDetailsUserDetailsService(
-				clientDetails);
-
-		return clientDetailsUserDetailsService;
-	}
-
-	@Bean
-	public WebResponseExceptionTranslator<OAuth2Exception> exceptionTranslator() {
-		return new CustomOAuth2ExceptionTranslator();
-	}
+	private static final String AUTHORITIES_CLAIM = "authorities";
 
 	/**
 	 * Class for configuring the OAuth resource server security
 	 */
 	@Configuration
-	@EnableResourceServer
-	@ComponentScan(basePackages = "ca.corefacility.bioinformatics.irida.repositories.remote")
-	@Order(Ordered.HIGHEST_PRECEDENCE + 2)
-	protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+	protected static class ResourceServerConfig {
 
-		@Autowired
-		private ResourceServerTokenServices tokenServices;
+		@Value("${server.base.url}")
+		private String serverBase;
 
 		@Autowired
-		private WebResponseExceptionTranslator<OAuth2Exception> exceptionTranslator;
+		OAuth2AuthorizationService authorizationService;
 
-		@Override
-		public void configure(final ResourceServerSecurityConfigurer resources) {
-			resources.resourceId("NmlIrida").tokenServices(tokenServices);
-			forceExceptionTranslator(resources, exceptionTranslator);
-		}
-
-		@Override
-		public void configure(final HttpSecurity httpSecurity) throws Exception {
-			httpSecurity.antMatcher("/api/**").authorizeRequests().regexMatchers(HttpMethod.GET, "/api.*")
-					.access("#oauth2.hasScope('read')").regexMatchers("/api.*")
-					.access("#oauth2.hasScope('read') and #oauth2.hasScope('write')");
-			httpSecurity.antMatcher("/api/**").headers().frameOptions().disable();
-			httpSecurity.antMatcher("/api/**").csrf()
-					.requireCsrfProtectionMatcher(new AntPathRequestMatcher("/api/oauth/authorize")).disable();
-			httpSecurity.antMatcher("/api/**").csrf().disable();
+		@Autowired
+		JwtDecoder jwtDecoder;
 
+		@Bean
+		@Order(Ordered.HIGHEST_PRECEDENCE + 2)
```

Comment:
`Ordered.HIGHEST_PRECEDENCE` is provided by Spring core.

Reply:
Okay. That makes sense. Is there anything wrong in using `HIGHEST_PRECEDENCE` here? Or do other services already use `HIGHEST_PRECEDENCE` and so we need to go even higher in precedence for this bit of code? Could we lower the precedence of those other services instead?

My thought process, by the way, is that if later on someone wants to set the precedence of loading a bean, they may set it to `HIGHEST_PRECEDENCE` and mistakenly think this means it will be loaded first, when it wont.

Alternatively, could we make a `SECURITY_PRECEDENCE` variable or `KEYSTORE_PRECEDENCE` or something and use that in place of `Ordered.HIGHEST_PRECEDENCE + 2`?

Reply:
`HIGHEST_PRECEDENCE + 2` actually lowers the precedence. What these precedences do is set `AuthorizationServerConfig.authorizationServerSecurityFilterChain` to apply first, then `UISecurityConfig.filterChain` and finally `ResourceServerConfig.resourceServerSecurityFilterChain`. This keeps the same behaviour/ordering as it was prior to the library change and refactor.

Reply:
Oh, okay. I didn't realize that higher precedence meant lower numbers. Could you maybe add a comment about that here (Add 2 to lower precedence)?

Otherwise, it looks fine.